### ISSUE TITLE
d-portal struggles with various parts of result data

### ISSUE THREAD

**Issue by stevieflow** [devinit/D-Portal#443]
Here's a screenshot of the sample XML ([published on the IATI standard documentation](http://iatistandard.org/203/activity-standard/example-xml/)), via d-portal -- specifically the results element:

![iati-example-d-portal](https://user-images.githubusercontent.com/1209944/36802258-5b39f982-1cac-11e8-94da-d8a754a15a14.png)

It looks like there's a few nesting issues in this.  The XML seems to include one result, with one indicator, but then multiple documents and multiple references.

d-portal seems to ingest this data, but then the layout is a bit broken, I think

**Comment by notshi** [comment]
Thanks for flagging this.

This looks like the new 2.03 result data standard which d-portal needs to update to where previously documents and links are not supported.

Could you send over the link to this particular example? It will help greatly in fixing the issue.

**Comment by stevieflow** [comment]
Thanks @notshi 

Here's the visualisation of the [2.02 example data](http://iatistandard.org/202/activity-standard/example-xml/)

![202](https://user-images.githubusercontent.com/1209944/36804743-61d9f4ca-1cb3-11e8-9402-187a28f420ee.png)

So, yes - something to do with the multiple inclusion of documents

Ill raise a new tickt about multiple indicators and vocab URIs

**Comment by notshi** [comment]
@stevieflow  Thanks for flagging these issues with examples and screenshots - they have been very helpful indeed!

I've now updated SAVi to support multiple references, links and documents for the 2.03 version. These changes are live on d-portal (as well as preview).

![new_result](https://user-images.githubusercontent.com/1515961/37047279-5c7fa5b6-2162-11e8-98c0-2cb867c64e36.png)


Do have a look through and see if it looks ok. These new changes should not affect older versions of the standard too much but if it does, do tell us.

Many thanks for your patience.

**Comment by stevieflow** [comment]
Hi @notshi - thanks for this

The layout looks fixed, great

One question I have - where did you get the names for the reference vocabularies.  The [codelist only includes a name for 99](http://iatistandard.org/203/codelists/ResultVocabulary/) - just interested in terms of where you got the WHO Registry and HIV/AIDS Indicator Registry name from ...?

I think @andylolz might be interested in this too!

**Comment by andylolz** [comment]
> One question I have - where did you get the names for the reference vocabularies. The [codelist only includes a name for 99](http://iatistandard.org/203/codelists/ResultVocabulary/) - just interested in terms of where you got the WHO Registry and HIV/AIDS Indicator Registry name from ...?

@stevieflow: There’s a [`result/reference`](http://iatistandard.org/203/activity-standard/iati-activities/iati-activity/result/reference/), that uses the [Result Vocabulary](http://iatistandard.org/203/codelists/ResultVocabulary/):

> A reference element to allow for the coded identification of a results framework.

…and a [`result/indicator/reference`](http://iatistandard.org/203/activity-standard/iati-activities/iati-activity/result/indicator/reference/), that uses the [Indicator Vocabulary](http://iatistandard.org/203/codelists/IndicatorVocabulary/):

> A standardised means of identifying the indicator from a code in a recognised vocabulary. Multiple vocabularies may be specified, but each vocabulary may be specified only once for each indicator.

This is the latter, so d-portal finds the WHO Registry and HIV/AIDS Indicator Registry on that codelist.

**Comment by stevieflow** [comment]
Thanks @andylolz - interesting

> Here, we’re dealing with indicators, so d-portal finds the WHO Registry and HIV/AIDS Indicator Registry on the Indicator Vocabulary codelist.

I'm not 100% we are.  In the [example XML](http://iatistandard.org/203/activity-standard/example-xml/) it *seems* that at the [results level](http://iatistandard.org/203/activity-standard/iati-activities/iati-activity/result/reference/), we have these reference vocabularies:

```
<!--result-reference starts-->
   <reference vocabulary="1" code="3429" />
   <reference vocabulary="7" code="861" />
   <reference vocabulary="99" code="B1" vocabulary-uri="http://example.com/indicators.html" />
<!--result-reference ends-->
```
So, vocabs 1 & 7 [don't exist on the ResultVoucabulary](http://iatistandard.org/203/codelists/ResultVocabulary/) list?

As you point out, at the indicator level they are fine....

**Comment by andylolz** [comment]
@stevieflow: Good point!

There are two instances of this in the example. You’ve posted the first one (which I don’t think is the one we’re seeing on d-portal). Here’s the second one:

```xml
<!--result-indicator starts-->
   <indicator measure="1" ascending="1" aggregation-status="1">
     <title>
      <narrative>Indicator title</narrative>
     </title>
     <description>
      <narrative>Indicator description text</narrative>
     </description>
     <document-link format="application/vnd.oasis.opendocument.text" url="http:www.example.org/docs/result_indicator_en.odt">
      <title>
       <narrative>Results Indicator Report 2013</narrative>
      </title>
      <description>
        <narrative>Report of results indicator</narrative>
      </description>
      <category code="A01" />
      <language code="en" />
      <document-date iso-date="2014-02-05" />
     </document-link>
     <reference vocabulary="1" code="3429" />
     <reference vocabulary="7" code="861" />
     <reference vocabulary="99" code="B1" indicator-uri="http://example.com/indicators.html" />
```

^^ These references are at indicator level, and I think these are the ones we’re seeing on d-portal.

So I would say this is a problem with the example XML, which (in the example you posted) is using vocabularies that aren’t on the [Result Vocabulary](http://iatistandard.org/203/codelists/ResultVocabulary/) codelist. It’s a `complete="1"` codelist, so I don’t think the example is actually even valid here!

Somewhat related: I don’t find it terribly helpful that the example XML uses strings like “A description that qualifies the activity taking place at the location.” An actual (invented) real world example seems like it would be more helpful for understanding what’s expected in these fields.

**Comment by andylolz** [comment]
@stevieflow: I’ve made a github issue for it (IATI/IATI-Extra-Documentation#465).

**Comment by notshi** [comment]
Thanks, both.

I've updated the higher tier reference to only use the [Result Vocabulary](http://iatistandard.org/203/codelists/ResultVocabulary/). Was confused by the similar looking references in the example xml.

Now the two instances of references are using the Result and Indicator vocab codelist accordingly.

Updated screenshot:
![result_vocab](https://user-images.githubusercontent.com/1515961/37306364-5f7a517c-262f-11e8-96a5-92094dec7b6d.png)
